Formula/kfctl.rb

```diff
@@ -0,0 +1,28 @@
+class Kfctl < Formula
+  desc "Kubeflow command-line interface"
+  homepage "https://github.com/kubeflow/kfctl"
+  url "https://github.com/kubeflow/kfctl.git",
+      tag: "v1.1.0",
+      revision: "9a3621e2b2ce6d2baa5c78acccec5028c4e4cbe1"
+  license "Apache-2.0"
+
+  depends_on "go@1.13" => :build
+
+  def install
+    system "make", "build"
+    bin.install "bin/darwin/kfctl"
+
+    # Install bash completion
+    output = Utils.safe_popen_read("#{bin}/kfctl", "completion", "bash")
+    (bash_completion / "kfctl").write output
+
+    # Install zsh completion
+    output = Utils.safe_popen_read("#{bin}/kfctl", "completion", "zsh")
+    (zsh_completion / "kfctl").write output
+  end
+
+  test do
+    run_output = shell_output("#{bin}/kfctl version 2>&1")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This is similar to `kubectl` and that is the reason for to have contrived tests similar to kubectl https://github.com/Homebrew/homebrew-core/blob/cf898a530a161125d5d15146fc16f5074c6372dd/Formula/kubernetes-cli.rb

Reply:
That does have an additional test to test the functionality, this is not good enough

Reply:
OK, So do you have any suggestions? This is a CLI tool for API.

Reply:
There's suggestions in my initial comment

Reply:
I disagree with the recommendations. 
>If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- What is the point of tests that are dependent on the API when it is not available? 
- How is this different from kubectl? I am trying to follow similar to kubectl.

Reply:
> What is the point of tests that are dependent on the API when it is not available?

They test a functional part of the application.

> How is this different from kubectl? I am trying to follow similar to kubectl.

It's not the same tool